### DISCUSSION TITLE
After upgrade from 0.61.1 to 0.80.1, some content has been lost

### DISCUSSION THREAD

**Discussion by alexisLefebvre** [outline/outline#7650] [Self Hosting]
We use outline through the Docker image `outlinewiki/outline:0.61.`.

We cloned this instance and upgraded to `outlinewiki/outline:0.77.3` then `outlinewiki/outline:0.80.1` in order to switch to the local storage instead of AWS (we were able to transfer the files from AWS to the server).

We were able to view and edit content in the new version.

However:
- some content has been lost, like bullet lists that had 3 elements now have only one
- some links are now displayed as iframe (many are empty since the websites block this)
  - if it was in a sentence, the accompanying text has been lost

---

What did we miss? Is there an upgrade step that we had to do? I can't find any documentation about updating the Docker version of Outline, like updating to version X, then Y and Z.

**Comment by tommoor** [comment]
Honestly not sure, I'm not aware of any changes that would cause data loss but you upgraded 2.5 years of versions all at once so it's probably better to do it through smaller jumps of versions and try and pinpoint where things change.

**Reply by alexisLefebvre** [reply]
Thanks for the idea!

So I did the upgrade through every version from 0.61.1 to the last release and it was an epic journey.

I used Docker tags to get each release: https://hub.docker.com/r/outlinewiki/outline/tags?name=0.63

For some reason the search function on GitHub releases doesn't work, e.g. when looking for `v0.62`: https://github.com/outline/outline/releases?q=v0.62&expanded=true

## `0.62.0`

I had to run the migrations manually after the upgrades: https://github.com/outline/outline/discussions/3190#discussioncomment-2316846

## `0.67.2`

Then starting from `0.67.2`, `yarn sequelize:migrate --env=production-ssl-disabled` must be replaced with `yarn db:migrate --env production-ssl-disabled` and one error appear:

```shell
yarn run v1.22.18
$ sequelize db:migrate --env production-ssl-disabled

Sequelize CLI [Node: 16.14.2, CLI: 6.4.1, ORM: 6.20.1]

Loaded configuration file "server/config/database.json".
Using environment "production-ssl-disabled".
== 20221111171828-fix-user-constraints: migrating =======

ERROR: insert or update on table "users" violates foreign key constraint "users_teamId_fkey"

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I had to update the team of 2 users:

```
UPDATE users SET "teamId" = 'ef4e6cf1-[…]' WHERE "teamId" = '330e03c8-[…]';
UPDATE 2
```

## `0.69.2`

When reaching `0.69.2`, the order of the sections of each page is changed, one section that was at the end is now in the middle.

It looks like this problem, even if we are still on the previous version (relatively to that issue):

- https://github.com/outline/outline/issues/5649

The history of the doc shows that I updated it, even if I didn't, it's like the live editor had an issue.

Anyway I can just rollback this change, and continue the updates.

## `0.75.2`

When reaching `0.75.2`, the same error occurs: the content of documents is duplicated just by looking at it.

## Lessons learned

- viewing the documents broke some of them
- but we didn't lost content like in the first attempt

So I will do the migration again with a new plan: 

1. Logout and close my Outline tabs to avoid issues with unexpected updates of documents
2. In our current instance, fix the issue with one user that is associated to the wrong team (to avoid the issue about the migration later)
3. Stop the new instance of Outline, delete all the data and get back to `0.61.1`
4. Export data from the current instance and import it in the new instance
5. Start the new instance
6. Upgrade the new instance to `0.66.3` and run `yarn sequelize:migrate --env=production-ssl-disabled`
7. Upgrade the new instance to `0.80.1` and run `yarn db:migrate --env production-ssl-disabled`
8. Check that the new instance works perfectly
9. Shut down the current instance and use the new instance instead
